os_migrate/tests/unit/test_server.py

```diff
@@ -259,23 +260,14 @@ def test_block_device_mapping_boot_disk_nocopy(self):
         ])
         self.assertTrue('image_id' not in sdk_params)
 
-    # TODO: add testcase for no data copy, no boot copy, load from optional image
     def test_block_device_mapping_boot_disk_nocopy_data_copy_false_no_additional_volumes_boot_image(self):
         srv = Server.from_sdk(None, sdk_server())
         srv.migration_params()['data_copy'] = False
         sdk_params = srv.sdk_params(None)
+        sdk_params["image_id"] = None
 
-        srv.update_sdk_params_block_device_mapping_nocopy(sdk_params)
-        self.assertEqual(sdk_params['block_device_mapping'], [
-            {
-                'boot_index': 0,
-                'uuid': 'uuid-test-image',
-                'delete_on_termination': False,
-                'destination_type': 'local',
-                'source_type': 'image',
-            }
-        ])
-        self.assertEqual(sdk_params['image_id'], 'uuid-test-image')
+        with self.assertRaises(exc.InconsistentState):
+            srv.update_sdk_params_block_device_mapping_nocopy(sdk_params)
```

Comment:
IMO it wold make sense to have both of these tests -- both for when image is specified and when nothing is specified and we need to raise an error. Now that the removal of 'image_id' from sdk_params is fixed, i think the original unit test shouldn't fail anymore.

![image](https://github.com/os-migrate/os-migrate/assets/331840/aad1f2bd-b3fe-4cb0-ba98-496451eb20d2)